packages/flutter_tools/lib/src/commands/upgrade.dart

```diff
@@ -245,16 +248,17 @@ class UpgradeCommandRunner {
       final String errorString = e.toString();
       if (errorString.contains('fatal: HEAD does not point to a branch')) {
         throwToolExit(
-          'You are not currently on a release branch. Use git to '
-          "check out an official branch ('stable', 'beta', 'dev', or 'master') "
-          'and retry, for example:\n'
-          '  git checkout stable'
+          'Unable to upgrade Flutter: HEAD does not point to a branch (Are you '
+          'in a detached HEAD state?).\n'
```

Comment:
The first message is better. I agree we should point users away from using `git` directly, but then terminology like `Flutter: HEAD` is just going to be confusing.

Reply:
How about this?
```
Unable to upgrade Flutter: Your Flutter checkout is currently not on a release branch (Are you
in a detached HEAD state?)...
```
I would prefer to keep the comment within `()` to suggest users who could diagnose this error with `git`.

Reply:
i think if they know what they're doing, "your flutter checkout is currently not on a release branch" will let them know what's happening. and if not, they can always just run the `flutter channel` command.

Reply:
Sounds reasonable. Likewise, I could edit the other message:
```
Unable to upgrade Flutter: The current Flutter branch/channel is not tracking any remote repository...
```
If it is okay to lose some verbosity in these messages, I will go ahead and make the changes.

Reply:
That looks good to me.